agent-ovs/lib/FSFaultSource.cpp

```
@@ -0,0 +1,138 @@
+    static const std::string EP_GROUP("endpoint-group-name");
+    static const std::string POLICY_SPACE_NAME("policy-space-name");
+    static const std::string EG_POLICY_SPACE("eg-policy-space");
+    static const std::string EP_GROUP_NAME("endpoint-group-name");
+    static const std::string FS_SEVERITY("severity");
+    static const std::string FS_DESCRIPTION("description");
+    static const std::string FS_CODE("faultCode");
+ 
+    try {
+        using boost::property_tree::ptree;
+        ptree properties;
+        Fault newfs;
+        typedef modelgbp::fault::SeverityEnumT SevEnum;
+
+        string pathstr = filePath.string();
+        read_json(pathstr, properties);
+        string fault_uuid = properties.get<string>(FS_UUID, "not_avaialable");
+        if (fault_uuid == "not_avaialable"){
+           LOG(ERROR) << "Fault UUID Unknown";
+           return;
+        } else { newfs.setFSUUID(fault_uuid); }
+ 
+        optional<string> ep_uuid = properties.get_optional<string>(EP_UUID);
+        if(ep_uuid){
+           newfs.setEPUUID(ep_uuid.get());
+        }
+         
+        static const string severity = properties.get<string>(FS_SEVERITY, "not_available");
+        if(severity == "critical"){
+           newfs.setSeverity(SevEnum::CONST_CRITICAL);
+        }else if(severity == "major"){
+           newfs.setSeverity(SevEnum::CONST_MAJOR);
+        }else if(severity == "minor"){  
+           newfs.setSeverity(SevEnum::CONST_MINOR);
+        }else if(severity == "info"){
+          newfs.setSeverity(SevEnum::CONST_INFO);
+        }else if(severity == "warning"){
+          newfs.setSeverity(SevEnum::CONST_WARNING);
+        }else if(severity == "cleared"){
+          newfs.setSeverity(SevEnum::CONST_CLEARED);
+        }else if(severity == "not_available"){
+          LOG(ERROR) << "Fault Severity unknown";
+          return;
+        }
+        
+        string description = properties.get<string>(FS_DESCRIPTION, "not_available");
```

same comments as above. don't provide a default value and let it throw an exception if the value is not provided